llvm/test/Transforms/PGOProfile/thinlto_indirect_call_promotion.ll

```diff
@@ -1,39 +1,45 @@
-; Do setup work for all below tests: generate bitcode and combined index
-; RUN: opt -module-summary %s -o %t.bc
-; RUN: opt -module-summary %p/Inputs/thinlto_indirect_call_promotion.ll -o %t2.bc
+; The raw profiles and reduced IR inputs are generated from Inputs/update_icall_promotion_inputs.sh
```

Comment:
Can we create a `.proftext` file and use it for this test instead of generating it from the script above?

Reply:
No because the point of this test is to catch regressions if the separator character (which is hardcoded in the proftext) changes again in the future.

Reply:
Using a `.proftext` file (and convert it to indexed profiles in the test using `RUN` line) generally has better readability.

For better test coverage, I changed to store raw profile data in `Inputs/` directory and added some comments at the beginning of test case Transforms/PGOProfile/thinlto_indirect_call_promotion.ll

Reply:
The script and c++ source is provided for on-demand update (e.g., version bump, etc). IIUC the goal of a `compiler-rt` test is to generate raw profiles from C++ source for each run of the IR test (`Transforms/PGOProfile/thinlto_indirect_call_promotion.ll`) (i.e.,the IR test would use the the output from the `compiler-rt` test). 

Practically, is there a way to run two tests one after the other so one could make use of the output of the other? Or shall the `compiler-rt` test verify the content of (raw or indexed) profile content?   (I'm assuming the IR test itself stays in `llvm/test/Transforms/PGOProfile` directory makes more sense here)

Reply:
The reason for having this in a single test this way is specifically to detect mismatches in the separator used in the PGO name and in the ThinLTO index. For this we need to convert from raw profile and feed it through ThinLTO+ICP. 

Having separate tests would not do this. I.e. with the change from ':' to ';', the compiler-rt test would presumably have failed and would be updated, but the test using the proftext input would not fail. The lack of testing these pieces together led to the prior separator rename not updating all the necessary compiler bits.